pkgs/development/compilers/llvm/8/clang/clang_qtcreator_support.patch

```diff
@@ -0,0 +1,334 @@
+diff --git a/tools/driver/CMakeLists.txt b/tools/driver/CMakeLists.txt
+index 89a3aa3c..433fb0d2 100644
+--- a/tools/driver/CMakeLists.txt
++++ b/tools/driver/CMakeLists.txt
+@@ -47,6 +47,9 @@ target_link_libraries(clang
+   clangFrontend
+   clangFrontendTool
+   clangSerialization
++  clangTidyPlugin
++  clangIncludeFixerPlugin
++  clazyPlugin
+   )
+ 
+ if(WIN32 AND NOT CYGWIN)
+diff --git a/tools/driver/driver.cpp b/tools/driver/driver.cpp
+index 51143e3d..426d6631 100644
+--- a/tools/driver/driver.cpp
++++ b/tools/driver/driver.cpp
+@@ -512,3 +512,18 @@ int main(int argc_, const char **argv_) {
+   // failing command.
+   return Res;
+ }
++
++// This anchor is used to force the linker to link the clang-tidy plugin.
++extern volatile int ClangTidyPluginAnchorSource;
++static int LLVM_ATTRIBUTE_UNUSED ClangTidyPluginAnchorDestination =
++    ClangTidyPluginAnchorSource;
++
++// This anchor is used to force the linker to link the clang-include-fixer
++// plugin.
++extern volatile int ClangIncludeFixerPluginAnchorSource;
++static int LLVM_ATTRIBUTE_UNUSED ClangIncludeFixerPluginAnchorDestination =
++    ClangIncludeFixerPluginAnchorSource;
++
++// This anchor is used to force the linker to link the clazy plugin.
++extern volatile int ClazyPluginAnchorSource;
++static int LLVM_ATTRIBUTE_UNUSED ClazyPluginAnchorDestination = ClazyPluginAnchorSource;
+diff --git a/tools/extra/CMakeLists.txt b/tools/extra/CMakeLists.txt
+index 6645b533..ad32042d 100644
+--- a/tools/extra/CMakeLists.txt
++++ b/tools/extra/CMakeLists.txt
+@@ -13,6 +13,7 @@ endif ()
+ add_subdirectory(clang-apply-replacements)
+ add_subdirectory(clang-reorder-fields)
+ add_subdirectory(modularize)
++add_subdirectory(clazy)
+ add_subdirectory(clang-tidy)
+ add_subdirectory(clang-tidy-vs)
+ 
+diff --git a/tools/extra/clazy/CMakeLists.txt b/tools/extra/clazy/CMakeLists.txt
+index d61714d0..f563976f 100644
+--- a/tools/extra/clazy/CMakeLists.txt
++++ b/tools/extra/clazy/CMakeLists.txt
```

Comment:
You almost completely replace `clazy`s original CMake file with this. Where does it come from and why is it needed?

Reply:
We need to build clazy as clang plugin, not stand alone version, to be later invoked from clang executable.

Reply:
Okay, but where do modified file come from? Qt Creator's fork? It might be a good idea to explain it as a comment in expression along with a link to the original patch (which doesn't change this `CMakeLists.txt`).

Reply:
I wrote it myself (actually just deleted all calls that is not needed for plugin build)